packages/runtime/test/src/runtime.ts

```diff
@@ -1,93 +1,69 @@
+      payFlags: { totalFee: 1000 }
+    };
   });
 
-  it("should verify only for lsig address(escrow account)", () => {
-    const lsig = runtime.getLogicSig(getProgram(programName), []);
-
-    let result = lsig.verify(lsig.address());
-    assert.equal(result, true);
-
-    result = lsig.verify(john.address);
-    assert.equal(result, false);
-  });
-});
-
-describe("Multi-Signature Test", () => {
-  useFixture("multi-signature");
-  const alice = new StoreAccount(10);
-  const john = new StoreAccount(100);
-  const bob = new StoreAccount(1000);
-
-  const runtime = new Runtime([alice, john, bob]);
-  // Generate multi signature account hash
-  const addrs = [alice.address, john.address, bob.address];
-  const mparams = {
-    version: 1,
-    threshold: 2,
-    addrs: addrs
-  };
-  const multsigaddr = multisigAddress(mparams);
-
-  it("should verify if threshold is verified and sender is multisigAddr", () => {
-    const lsig = runtime.getLogicSig(getProgram(multiSigProg), []);
-    // lsig signed by alice
-    lsig.sign(alice.account.sk, mparams);
-    // lsig signed again (threshold = 2) by john
-    lsig.appendToMultisig(john.account.sk);
+  it("should execute the lsig and verify john(delegated signature)", () => {
+    lsig.sign(john.account.sk);
+    runtime.executeTx(txnParam, getProgram(programName), []);
 
-    const result = lsig.verify(multsigaddr);
-    assert.equal(result, true);
+    // balance should be updated because logic is verified and accepted
+    const bobAcc = runtime.getAccount(bob.address);
+    assert.equal(bobAcc.balance(), minBalance + 1000);
   });
 
-  it("should not verify if threshold is achieved but sender is not multisigAddr", () => {
-    const lsig = runtime.getLogicSig(getProgram(multiSigProg), []);
-    // lsig signed by alice
-    lsig.sign(alice.account.sk, mparams);
-    // lsig signed again (threshold = 2) by john
-    lsig.appendToMultisig(john.account.sk);
+  it("should not verify signature because alice sent it", () => {
+    txnParam.fromAccount = alice.account;
 
-    const result = lsig.verify(bob.address);
-    assert.equal(result, false);
+    // execute transaction (logic signature validation failed)
+    expectTealError(
+      () => runtime.executeTx(txnParam, getProgram(programName), []),
+      ERRORS.TEAL.LOGIC_SIGNATURE_VALIDATION_FAILED
+    );
   });
 
-  it("should not verify if threshold is not achieved but sender is multisigAddr", () => {
-    const lsig = runtime.getLogicSig(getProgram(multiSigProg), []);
-    // lsig signed by alice
-    lsig.sign(alice.account.sk, mparams);
-
-    const result = lsig.verify(multsigaddr);
-    assert.equal(result, false);
+  it("should verify signature but reject logic", async () => {
+    const logicSig = runtime.getLogicSig(getProgram("reject.teal"), []);
+    txnParam.lsig = logicSig;
+    txnParam.fromAccount = john.account;
+
+    logicSig.sign(john.account.sk);
+    // execute transaction (rejected by logic after signature validation)
```

Comment:
Please add more details why it's rejected (eg, because fromAccount check failed in the program)

Reply:
ok